Mathlib/RepresentationTheory/Basic.lean

```diff
@@ -290,7 +291,34 @@ theorem ofMulAction_single (g : G) (x : H) (r : k) :
 #align representation.of_mul_action_single Representation.ofMulAction_single
 
 end MulAction
+section DistribMulAction
 
+variable (M A : Type*) [Monoid M] [AddCommGroup A] [DistribMulAction M A]
+
+/-- Turns an `AddCommGroup` `A` with a `DistribMulAction` of a monoid `M` into a
+`ℤ`-linear `M`-representation on `A`. -/
+def ofDistribMulAction : Representation ℤ M A :=
+  (addMonoidEndRingEquivInt A : _ →* _).comp (DistribMulAction.toAddMonoidEnd M A)
+
```

Comment:
Would it also be useful to have a def of `Representation R M A` given some commutative ring R acting compatibly on M and A? Then maybe typeclass inference can fill in the actions if R=Z?

Reply:
Thanks, that works